src/main/java/edu/ie3/datamodel/io/source/sql/SqlWeatherSource.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.source.sql;
+
+import edu.ie3.datamodel.io.connectors.SqlConnector;
+import edu.ie3.datamodel.io.factory.timeseries.TimeBasedWeatherValueData;
+import edu.ie3.datamodel.io.factory.timeseries.TimeBasedWeatherValueFactory;
+import edu.ie3.datamodel.io.source.IdCoordinateSource;
+import edu.ie3.datamodel.io.source.WeatherSource;
+import edu.ie3.datamodel.models.timeseries.individual.IndividualTimeSeries;
+import edu.ie3.datamodel.models.timeseries.individual.TimeBasedValue;
+import edu.ie3.datamodel.models.value.WeatherValue;
+import edu.ie3.util.TimeUtil;
+import edu.ie3.util.interval.ClosedInterval;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.locationtech.jts.geom.Point;
+
+/** SQL source for weather data */
+public class SqlWeatherSource implements WeatherSource {
+  private static final Logger logger = LogManager.getLogger(SqlWeatherSource.class);
+
+  private static final String DEFAULT_TIMESTAMP_PATTERN = "yyyy-MM-dd HH:mm:ss.0";
```

Comment:
Is there a reason why you don't use the default pattern used in edu.ie3.util.TimeUtil? This might make sense for consistency ...?

Reply:
Same here